dash_docs/chapters/advanced_callbacks/index.py

```diff
@@ -149,12 +149,63 @@ def slow_function(input):
     '''),
 
     rc.Syntax(examples['callbacks-prevent-initial-call.py'][0]),
     rc.Example(examples['callbacks-prevent-initial-call.py'][1]),
 
+    rc.Markdown('''
+
+    However, the above behavior only applies if both the callback output and input are present in the app layout upon initial load of the application.
+
+    It is important to note that `prevent_initial_call` will not prevent a callback from firing in the case where the callback's input is inserted into the layout as the result of another callback after the app initially loads **unless the output is inserted alongside that input**!
+
+    In other words, if the output of the callback is already present in the app layout before its input is inserted into the layout, `prevent_initial_call` will not prevent its execution when the input is first inserted into the layout.
+
+    Consider the following example:
+
+    ```python
+      import dash
+      import dash_core_components as dcc
+      import dash_html_components as html
+      from dash.dependencies import Input, Output, State
+      import urllib
+      app = dash.Dash(__name__, suppress_callback_exceptions=True)
+      server = app.server
+      app.layout = html.Div([
+          dcc.Location(id='url'),
+          html.Div(id='layout-div'),
+          html.Div(id='content')
+      ])
+
+      @app.callback(Output('content', 'children'), Input('url', 'pathname'))
+      def display_page(pathname):
+          return html.Div([
+              dcc.Input(id='input', value='hello world'),
+              html.Div(id='output')
+          ])
+
+      @app.callback(Output('output', 'children'), Input('input', 'value'), prevent_initial_call=True)
+      def update_output(value):
+          print('>>> update_output')
+          return value
+
+      @app.callback(Output('layout-div', 'children'), Input('input', 'value'), prevent_initial_call=True)
+      def update_layout_div(value):
+          print('>>> update_layout_div')
+          return value
+
+    ```
+
+    In this case, `prevent_initial_call` will prevent the `update_output()` callback from firing when it's input is first inserted into the app layout as a result of the `display_page()` callback. This is because both the callback's input and output are already contained within the app layout when the callback is set to execute.
+
+    However, `prevent_initial_call` will not prevent the `update_layout_div()` callback from firing, because the callback's output is already present in the app layout but its input is not. Since `suppress_callback_exceptions=True` is specified in this example Dash app, Dash has to asssume that the input is present in the app layout when the app is initialized. This means that as far as the output element is concerned, the new input component is a change just like if the existing input had been given a new value, rather than its initial render.
```

Comment:
The last sentence is a little convoluted here, not sure if what I've suggested makes sense, but would revise in whatever way you feel makes the most sense.

The edits to the initial sentence are proposed to make it a little easier to follow what's happening here, but it works either way.

```suggestion
    However, because the app layout contains only the output of the callback, and not its input, `prevent_initial_call` will not prevent the `update_layout_div()` callback from firing. Since `suppress_callback_exceptions=True` is specified here, Dash has to assume that the input is present in the app layout when the app is initialized. From the perspective of the output element in this example, the new input component is handled as if an existing input had been provided a new value, rather than treating it as initially rendered.
```